### DISCUSSION TITLE
Return address record when saving an address instead of true or false

### DISCUSSION THREAD

**Discussion by aloco** [craftcms/commerce#2508] [Ideas] [answered]
Hi,

we have a custom controller which provides an action to save various user data, e.g. user first name, last name, and address data. We are returning JSON Data with the updated or created user data. We are using `saveAddress` of the Address Service, however, this method returns true or false instead of the address record which makes it very difficult for us to work with. Is there a way to store an address and retrieving at least the address id on success? Not sure how to use `EVENT_AFTER_SAVE_ADDRESS` to include the new address in the JSON Response of the Controller..

Thank you for your help!

**Comment by lukeholder** [comment, marked as answer]
All save methods in Craft and Craft Commerce services return true and false only at the moment.

What would you like returned? You still have reference to the address model that you passed to `saveAddress($address)` and if it was saved, it will have the new record ID added to it.

**Reply by aloco** [reply]
Hi Luke,

thank you for your reply, you are right we can receive the record id from the Model - wasn´t aware of this
